src/system/connectivity/wifi.cpp

```diff
@@ -1,12 +1,37 @@
 #include "wifi.h"
 #include "utils/logger.h"
-#ifdef ARDUINO_ESP8266_ESP12
-#include <ESP8266WiFi.h>
-#else
-#include <WiFi.h>
-#endif
+// #ifdef ARDUINO_ESP8266_ESP12
+// #include <ESP8266WiFi.h>
+// #else
+//#include <WiFi.h>
+//#endif
 
 static WiFiManager wm;
+using namespace WiFiConnection;
+
+    static String templateID;
+    static String deviceName;
+    static String token;
+
+static void saveParamCallback()
+{
+    if(wm.server->hasArg("template_id"))
+    {
+        templateID = wm.server->arg("template_id");
+        logger << LOG_INFO << "TemplateID = " << templateID << EndLine;
+    }
+    if(wm.server->hasArg("device_name"))
+    {
+        deviceName = wm.server->arg("device_name");
+        logger << LOG_INFO << "Device Name = " << deviceName << EndLine;
+    }
+    if(wm.server->hasArg("auth_token"))
+    {
+        token = wm.server->arg("auth_token");
+        logger << LOG_INFO << "Token = " << token << EndLine;
+    }
```

Comment:
Should we handle errors?

Reply:
maybe just prints, not sure what to do if one parameter is missing. any idea?